zulipterminal/ui_tools/boxes.py

```diff
@@ -491,6 +504,48 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                     self.model.save_draft(message)
                 elif message != saved_draft:
                     self.view.controller.save_draft_confirmation_popup(message)
+
+        # Since, there are two keys mapped to GO_RIGHT
+        # i.e. ['right', 'l'], we dont need 'l' key to satisfy
+        # the condition: is_command_key("GO_RIGHT", key).
+        # FIXME: Refactor this condition?
+        elif key == 'right':
+            header = self.header_write_box
+            if self.to_write_box is None:
+                if (self.focus_position == self.FOCUS_CONTAINER_HEADER
+                        and header.focus_col == self.FOCUS_HEADER_BOX_STREAM):
+                    cursor_pos = header[self.FOCUS_HEADER_BOX_STREAM].edit_pos
+                    stream_name = (header[self.FOCUS_HEADER_BOX_STREAM]
+                                   .edit_text)
+                    stream_name_len = len(stream_name)
+                    if stream_name_len == cursor_pos:
+                        if not self.model.is_valid_stream(stream_name):
+                            self.footer_notify_invalid_stream()
+                            return key
```

Comment:
This likely needs to handle the same setting of recipient ids and stream id as in the other block of very similar code - which is a good reason why this would be handled well with a refactor to handle the different keys together.

Similarly the similar block below, and another for PMs?